backend/telemetry_core/src/state/batched.rs

```diff
@@ -88,6 +140,16 @@ impl State {
 
     /// Drain updates for all feeds and return serializer.
     pub fn drain_updates_for_all_feeds(&mut self) -> FeedMessageSerializer {
+        self.metadata.update(&self.chains);
+        if let Some(path) = &self.metadata_path {
+            if let Err(err) = serde_json::to_vec(&self.metadata)
+                .map_err(anyhow::Error::from)
+                .and_then(|bytes| std::fs::write(path, bytes).map_err(anyhow::Error::from))
+            {
+                log::error!("Failed to save metadata: {err}");
+            }
+        }
```

Comment:
Ideally we'd check if it has changed before writing. If it didn't, no need to write an update.